src/server/vnparse.c

```diff
@@ -2432,6 +2434,55 @@ reliable_job_node_delete(pbs_list_head *node_list, char *nname)
 
 }
 
+/**
+ *
+ * @brief
+ *	Find an entry from the list of nodes for a job
+ *	named 'nname', and mark this node host as having
+ *	successfully executed execjob_prologue hook,
+ *	resulting in a hook event accept.
+ *	If no existing node host was matched, then add one.
+ *
+ * @param[in]	node_list	- the PBS node list
+ * @param[in]	nname		- node hostname to search for.
+ *
+ * @return reliable_job_node *
+ *
+ * @retval <reliable_nob_node entry>	- the updated/added node entry
+ * @retval NULL  			- if an error occurred.
+ * 
+ */
+reliable_job_node *
+reliable_job_node_set_prologue_hook_success(pbs_list_head *node_list, char *nname)
+{
+	reliable_job_node *rjn = NULL;
+
+	if ((node_list == NULL) || (nname == NULL))
+		return (NULL);
+
+	for (rjn = (reliable_job_node *)GET_NEXT(*node_list); rjn != NULL; rjn = (reliable_job_node *)GET_NEXT(rjn->rjn_link)) {
+		if (strcmp(rjn->rjn_host, nname) == 0) {
+			rjn->prologue_hook_success = 1;
+			return (rjn);
+		}
+	}
+	/* no entry matched so add one */
+	rjn = (reliable_job_node *)malloc(sizeof(reliable_job_node));
+	if (rjn == (reliable_job_node *)0) {
+		log_err(errno, __func__, msg_err_malloc);
+		return (NULL);
+	}
+	CLEAR_LINK(rjn->rjn_link);
+
+	strncpy(rjn->rjn_host, nname, sizeof(rjn->rjn_host) - 1);
```

Comment:
This won't help you.  You malloc()'d space for rjn.  This means rjn->rjn_host contains garbage.  This will make sure you don't write on the last byte of your buffer, but that last byte will contain garbage.  Either set a '\0' into the last byte, or use snprintf().

Reply:
I'll use snprintf.